grudge/op.py

```diff
@@ -47,38 +24,136 @@
 """
 
 
+from arraycontext.metadata import FirstAxisIsElementsTag
+from arraycontext.loopy import make_loopy_program
+from arraycontext.container.traversal import freeze
+
 from numbers import Number
-from pytools import memoize_on_first_arg
 
-import numpy as np  # noqa
+from functools import reduce
+
+from pytools import (
+    memoize_in,
+    memoize_on_first_arg,
+    keyed_memoize_in
+)
 from pytools.obj_array import obj_array_vectorize, make_obj_array
-import pyopencl.array as cla  # noqa
-from grudge import sym, bind
 
+from meshmode.dof_array import DOFArray
+
+import numpy as np
 import grudge.dof_desc as dof_desc
 
-from meshmode.mesh import BTAG_ALL, BTAG_NONE, BTAG_PARTITION  # noqa
-from meshmode.dof_array import freeze, flatten, unflatten
+from grudge.trace_pair import (  # noqa
+    interior_trace_pair,
+    interior_trace_pairs,
+    connected_ranks,
+    cross_rank_trace_pairs,
+    bdry_trace_pair,
+    bv_trace_pair
+)
```

Comment:
These are imported here so we can use `grudge.op` as the main source of core DG-related functions. I.e. `op.interior_trace_pairs`, `op.cross_rank_trace_pairs`, etc

Reply:
OK, sure. Maybe revisit for #99.